src/phpDocumentor/Compiler/Linker/DescriptorRepository.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @link http://phpdoc.org
+ */
+
+namespace phpDocumentor\Compiler\Linker;
+
+use phpDocumentor\Descriptor\ClassDescriptor;
+use phpDocumentor\Descriptor\DescriptorAbstract;
+use phpDocumentor\Descriptor\InterfaceDescriptor;
+use phpDocumentor\Descriptor\NamespaceDescriptor;
+use phpDocumentor\Descriptor\TraitDescriptor;
+
+final class DescriptorRepository
+{
+    private const CONTEXT_MARKER = '@context';
+
+    /** @var DescriptorAbstract[] */
+    private $elementList = [];
+
+    /**
+     * Attempts to find a Descriptor object alias with the FQSEN of the element it represents.
+     *
+     * This method will try to fetch an element after normalizing the provided FQSEN. The FQSEN may contain references
+     * (bindings) that can only be resolved during linking (such as `self`) or it may contain a context marker
+     * {@see CONTEXT_MARKER}.
```

Comment:
I think we need to check if this `CONTEXT_MARKER` is still working... because the `see` tag has now the requirement of a `FQSEN` which is always a valid element name. Maybe this is a broken tag :-)

Reply:
I'll write a test when I continue :)